test/assetGiveaway/fixtures.ts

```diff
@@ -0,0 +1,122 @@
+import {
+  ethers,
+  deployments,
+  getUnnamedAccounts,
+  getNamedAccounts,
+} from 'hardhat';
+import {expect} from '../chai-setup';
+import MerkleTree from '../../lib/merkleTree';
+import helpers from '../../lib/merkleTreeHelper';
+const {createDataArrayAssets} = helpers;
+
+const ipfsHashString =
+  '0x78b9f42c22c3c8b260b781578da3151e8200c741c6b7437bafaff5a9df9b403e';
+
+import {expectReceiptEventWithArgs, waitFor} from '../utils';
+
+export const setupGiveaway = async (
+  assetType: string,
+  mint?: boolean,
+  amount?: number,
+  supply?: number,
+  to?: string
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+): Promise<any> => {
+  const {
```

Comment:
The use of createFixture this way (creating a new one at each invocation of `setupGiveaway` will not reuse existing fixture

The pattern you want to use is as follow:

```
const setupGiveaway = deployments.createFixture(async function (hre, options) {
// the options field should be JSON strinigyable
// it will be passed as parameter by when calling `setupGiveaway`
// and if you group test using the same options, the fixtures will benefit from caching
});
```

so in your tests you can do as before, except you ll use an object so you can pass multiple options

```
const setUp = await setupGiveaway({assetTupe: 'test'});
```

Reply:
Updated